databroker/core.py

```diff
@@ -1578,11 +1578,19 @@ def _transpose(in_data, keys, field):
         for k in keys:
             out[k][j] = dd[k]
     for k in keys:
-        # compatibility with dask < 2
-        if hasattr(out[k][0], 'shape'):
-            out[k] = dask.array.stack(out[k])
-        else:
-            out[k] = dask.array.array(out[k])
+        try:
+            # compatibility with dask < 2
+            if hasattr(out[k][0], 'shape'):
+                out[k] = dask.array.stack(out[k])
+            else:
+                out[k] = dask.array.array(out[k])
+        except NotImplementedError:
```

Comment:
Interesting. The case that brought me here was ragged arrays, which numpy/dask have no real solution for and thus raise `NotImplementedError`. If we hit the code path you linked, we probably have a lying descriptor and should fix it with a transformer rather than pulling (potentially large) data into memory as a numpy array.

Reply:
OTOH, it might be clever to add an escape hatch here in case we are overwhelmed by transform-related issues during the reply. Something like, "If the environment variable `DATABROKER_ARRAY_FALLBACK` is set, catch `ValueError`, warn that things will be slow, and return a numpy array rather than failing." I don't think that should be on by _default_ though.

Reply:
Won't we hit the ValueError when we call dask.array.stack() with a ragged array?

Reply:
No, numpy treats a ragged array as an array of "objects" (which happen to be lists or variable length) and thus we hit https://github.com/dask/dask/blob/5f61f7f79b861be5ffdbf59950889166628a8fa6/dask/array/core.py#L2532-L2536

Reply:
Added emergency fallback in 6a04e03f